### ISSUE TITLE
Code suitability

### ISSUE THREAD

**Issue by wwjwy** [HiLab-git/DTC#1]
Can this code be directly used for the segmentation of 2D images?
I would be very grateful if you could reply

**Comment by Luoxd1996** [comment]
Hi wwjwy,
     Sorry, we just use this repo for 3D medical image segmentation, but it does not mean this can't be used for 2D image segmentation, you need to modify something for your 2D task, such as network, loss function, visualization, and so on.  For 2D image segmentation, you can combine DTC with our new repo SSL4MIS (https://github.com/HiLab-git/SSL4MIS).
Best wish,
Xiangde Luo.

**Comment by wwjwy** [comment]
您好：

非常感谢您的回复。 您的示例代码是十分优秀的工作，我相信这会对我有很大帮助。 通过您示例的指导，使用您提供的预处理ACDC数据集，我可以成功运行。
现在我想在我自己的数据集上运行，我的数据集是2D图片，有Image和对应的mask。我想请问该怎么制作成您提供的那种形式的预处理的数据集文件呢？希望可以得到您的帮助，谢谢！

王文君







xdluo <notifications@github.com> 于2020年10月30日周五 上午10:54写道：

**Comment by Luoxd1996** [comment]
Hi wwjwy,
I'm not sure which type of files are you want, but for SSL4MIS, the input of the network is Tensor, you can read your image and mask by OpenCV or PIL and convert it to the array and Tensor. For the h5py file, we just used it to save some Hard Disk memory, so you can choose data types based on your own situation, more details of h5py you can find at https://docs.h5py.org/en/stable/ .
Best wish,
Xiangde Luo.

**Comment by wwjwy** [comment]
Hi Luo:

我的意思是，我想使用您SSL4MIS项目中提供的代码对我自己的2D图片进行半监督语义分割训练。但是我需要把自己的数据集预处理成您提供的ACDC那种形式，这里我不知道该如何实现。
请问这里使用到的代码在SSL4MIS项目中有提供吗？

王文君


xdluo <notifications@github.com> 于2020年10月30日周五 上午11:33写道：

**Comment by Luoxd1996** [comment]
Hi wwjwy,
Yeah, I got it, I will update the pre-processing code later. Thanks for your question and suggestion.
Best wish,
Xiangde Luo.

**Comment by Luoxd1996** [comment]
Hi wwjwy,
I have updated the ACDC pre-processing code in [here](https://github.com/HiLab-git/SSL4MIS/blob/master/code/dataloaders/acdc_data_processing.py), there is a demo to save NIfTI data as .h5 file, may it can help you.
Best wish,
Xiangde Luo.

**Comment by wwjwy** [comment]
Thank you very much😁

xdluo <notifications@github.com> 于2020年10月30日周五 下午2:23写道：

**Comment by Luoxd1996** [comment]
Hi wwjwy,
Do you have any questions? if not, I will close this issue. Any questions, you can contact me at any time by email or new issues.
Best wish,
Xiangde Luo

**Comment by wwjwy** [comment]
I have no questions now. Thanks

xdluo <notifications@github.com> 于2020年10月30日周五 下午2:33写道：
